ldk/goja/src/types/oliveHelps/index.d.ts

```diff
@@ -158,32 +157,182 @@ declare namespace OliveHelps {
     all(cb: (whispers: Whisper[]) => void): void;
   }
 
-  interface NewWhisper {
-    label: string;
-    components: Component[];
+  enum Urgency {
+    ERROR = 'error',
+    NONE = 'none',
+    SUCCESS = 'success',
+    WARNING = 'warning',
   }
 
-  interface Whisper {
-    id: string;
+  interface Component {
+    type: string;
+    id?: string;
+  }
+
+  interface Button extends Component {
     label: string;
-    components: Component[];
+    onClick: () => void;
+    submit?: boolean;
+  }
 
-    onChange(cb: (whisper: Whisper) => void): void;
+  interface Checkbox extends Component {
+    label: string;
+    tooltip?: string;
+    value?: boolean;
+    onChange: (value: boolean) => void;
+  }
 
-    close(cb: () => void): void;
+  interface Link extends Component {
+    href?: string;
+    style?: Urgency;
+    onClick?: (value: string) => void;
+    text: string;
+    textAlign?: string;
   }
 
-  interface Component {
-    type: string;
-    id: string;
+  interface ListPair extends Component {
+    copyable: boolean;
+    label: string;
+    style?: string;
+    value?: string;
   }
 
-  interface MarkdownComponent extends Component {
+  interface Markdown extends Component {
     body: string;
   }
 
-  interface ButtonComponent extends Component {
+  interface Message extends Component {
     body: string;
+    header: string;
+    style?: string;
+    textAlign?: string;
+  }
+
+  interface NumberInput extends Component {
+    label: string;
+    max?: number;
+    min?: number;
+    onChange?: (value: string) => void;
+    step?: number;
+    tooltip?: string;
+    value?: number;
+  }
+
+  interface Password extends Component {
+    label: string;
+    onChange?: (value: string) => void;
+    tooltip?: string;
+    value?: string;
+  }
+
+  interface RadioGroup extends Component {
+    onSelect: (value: number) => void; 
+    options: string[];
+    selected?: number;
+  }
+
+  interface Select extends Component {
+    label: string;
+    onSelect: (value: number) => void;
+    tooltip?: string;
+    options: string[];
+  }
+
+  interface Telephone extends Component {
+    label: string;
+    onChange?: (value: string) => void;
+    pattern?: RegExp;
+    tooltip?: string;
+    value?: string;
+  }
+
+  interface TextInput extends Component {
+    label: string;
+    onChange?: (value: string) => void;
+    tooltip?: string;
+    value?: string;
+  }
+
+  interface CollapseBox extends Component {
+    children: Array<
+      | Button
```

Comment:
It looks like this union type is shared; you can declare a type alias:

```
export type ChildrenElement = Button | Checkbox | Link | ListPair | ...
```

and then consume it like you would any other type:

```
interface CollapseBox extends Component {
...
children: ChildrenElement[];
...
}
```